packages/lib/src/interactions/InteractionsProvider.tsx

```diff
@@ -36,29 +37,63 @@ function InteractionsProvider(props: { children: ReactNode }) {
   );
 
   const shouldInteract = useCallback(
-    (id: string, event: MouseEvent) => {
-      const registeredKeys = [...interactionMap.values()]
-        .map((params: Interaction) => params.modifierKey)
-        .filter(isDefined);
+    (interactionId: string, event: MouseEvent | WheelEvent) => {
+      const registeredInteractions = [...interactionMap.entries()].map(
+        ([id, interaction]) => ({
+          id,
+          ...interaction,
+        })
```

Comment:
Perhaps this is the data structure we should use for the `interactionMap` state? It'd be better to loop through the array when registering/unregistering interactions rather than on every interaction event.